shared/transforms/shared_shorthand2xccdf.xslt

```diff
@@ -68,117 +68,27 @@
-            </xsl:otherwise>
-          </xsl:choose>
-          <xsl:choose>
-            <xsl:when test="contains(conflicts/@prodtype, $prod_type)">
-              <xsl:apply-templates select="conflicts[contains(@prodtype, $prod_type)]"/>
-            </xsl:when>
-            <xsl:when test="conflicts/@prodtype = 'all'">
-              <xsl:apply-templates select="conflicts[@prodtype = 'all']"/>
-            </xsl:when>
-            <xsl:otherwise>
-              <xsl:apply-templates select="conflicts[not(@prodtype)]"/>
-            </xsl:otherwise>
-          </xsl:choose>
-          <xsl:choose>
-            <xsl:when test="contains(platform/@prodtype, $prod_type)">
-              <xsl:apply-templates select="platform[contains(@prodtype, $prod_type)]"/>
-            </xsl:when>
-            <xsl:when test="platform/@prodtype = 'all'">
-              <xsl:apply-templates select="platform[@prodtype = 'all']"/>
-            </xsl:when>
-            <xsl:otherwise>
-              <xsl:apply-templates select="platform[not(@prodtype)]"/>
-            </xsl:otherwise>
-          </xsl:choose>
-          <xsl:choose>
-            <xsl:when test="contains(ident/@prodtype, $prod_type)">
-              <xsl:apply-templates select="ident[contains(@prodtype, $prod_type)]"/>
-            </xsl:when>
-            <xsl:when test="ident/@prodtype = 'all'">
-              <xsl:apply-templates select="ident[@prodtype = 'all']"/>
-            </xsl:when>
-            <xsl:otherwise>
-              <xsl:apply-templates select="ident[not(@prodtype)]"/>
-            </xsl:otherwise>
-          </xsl:choose>
+          <xsl:apply-templates select="title[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="title[not(@prodtype)]"/>
+          <xsl:apply-templates select="description[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="description[not(@prodtype)]"/>
+          <xsl:apply-templates select="warning[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="warning[not(@prodtype)]"/>
+          <xsl:apply-templates select="ref[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="ref[not(@prodtype)]"/>
+          <xsl:apply-templates select="rationale[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="rationale[not(@prodtype)]"/>
+          <xsl:apply-templates select="requires[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="requires[not(@prodtype)]"/>
+          <xsl:apply-templates select="conflicts[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="conflicts[not(@prodtype)]"/>
+          <xsl:apply-templates select="platform[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="platform[not(@prodtype)]"/>
+          <xsl:apply-templates select="ident[contains(@prodtype, $prod_type)]"/>
+          <xsl:apply-templates select="ident[not(@prodtype)]"/>
```

Comment:
if you merge some of the logic into a single select it will be way more readable. especially the logic of attr is not there or if it's there it's exactly XYZ.

Reply:
Logic merged, thank you for the improvement suggestion.